app/controllers/Application.scala

```diff
@@ -117,15 +117,21 @@ object Application extends Controller {
     ).getOrElse(NotFound(views.html.static.trouble("No such scheme: " + scheme_id)))
   }
 
-  def topicSubscribe(id: Int) = Action { implicit request => //= mustAuthenticate { username => implicit request =>
+  def topicSubscribe(id: Int, cancel: Boolean) = Action { implicit request => //= mustAuthenticate { username => implicit request =>
     Topic.findById(id).map( topic =>
-      topicSubIfSubscriber(User.findByName("richard").get, topic)
+      topicSubIfSubscriber(User.findByName("richard").get, topic, cancel)
     ).getOrElse(NotFound(views.html.static.trouble("No such topic: " + id)))
   }
 
-  private def topicSubIfSubscriber(user: User, topic: Topic)(implicit request: Request[AnyContent]): Result = {
-    Subscriber.findByUserId(user.id).map( sub =>
-      Redirect(routes.Application.topic(topic.id))
+  private def topicSubIfSubscriber(user: User, topic: Topic, cancel: Boolean)(implicit request: Request[AnyContent]): Result = {
```

Comment:
Maybe `topicSubManage` or `manageTopicSubscription`? Or something else... as this also handles unsubscribing from a topic, I think the name `topicSubIfSubscriber` could be confusing.